primitives/src/main/kotlin/it/unibo/collektive/codgen/utils/Utils.kt

```diff
@@ -0,0 +1,234 @@
+package it.unibo.collektive.codgen.utils
+
+import com.squareup.kotlinpoet.AnnotationSpec
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.MemberName.Companion.member
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.TypeVariableName
+import com.squareup.kotlinpoet.asTypeName
+import kotlin.math.pow
+import kotlin.reflect.KCallable
+import kotlin.reflect.KFunction
+import kotlin.reflect.KProperty
+
+private val FIELD_INTERFACE = ClassName("it.unibo.collektive.field", "Field")
```

Comment:
Make this module depend on DSL and use a reference to the field class, so that we are robust to refactorings

Reply:
Done